tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ManagedIdentityTests.cs

```diff
@@ -0,0 +1,189 @@
+        [DataRow(AzureResource.webapp, Mi_res_id, DisplayName = "ResourceID Web App")]
+        [DataRow(AzureResource.function, Mi_res_id, DisplayName = "ResourceID Function App")]
+        public async Task AcquireMSITokenAsync(AzureResource azureResource, string userIdentity)
+        {
+            //Arrange
+            AuthenticationResult result = null;
+
+            //Set the Environment Variables
+            bool isEnvironmentVariableSet = await SetEnvironmentVariablesAsync(azureResource)
+                .ConfigureAwait(false);
+
+            if (!isEnvironmentVariableSet)
+            {
+                Assert.Fail("AcquireMSITokenForWebAppAsync failed to set environment variables.");
+            }
+
+            //form the URI 
+            string uri = s_baseURL + $"GetMSIToken?azureresource={azureResource}&uri=";
+
+            //Create CCA with Proxy
+            IConfidentialClientApplication cca = CreateCCAWithProxy(uri);
+
+            //Act
+            try
+            {
+                result = await cca.AcquireTokenForClient(s_msi_scopes)
+                    .WithManagedIdentity(userIdentity)
+                    .ExecuteAsync().ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Trace.WriteLine(ex);
+                Assert.IsNull(ex.Message);
+            }
+
+            //Assert
+            Assert.IsNotNull(result.AccessToken);
+            Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+
+            //User Identity Client ID 
+            //Need to check if we are exposing Client ID in msal token response
+            //if (!string.IsNullOrEmpty(userIdentity))
+            //{
+            //    Assert.AreEqual(UserAssignedClientID, result.);
+            //}
+
+            result = await cca.AcquireTokenForClient(s_msi_scopes)
+                .WithManagedIdentity(userIdentity)
+                .ExecuteAsync().ConfigureAwait(false);
+
+            Assert.IsNotNull(result.Scopes);
+            Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
+
+            ClearEnvironmentVariables();
+
+        }
+
+        private async Task<bool> SetEnvironmentVariablesAsync(AzureResource resource)
+        {
+            //Get the Web App Environment Variables from the MSI Helper Service
+            string uri = s_baseURL + "GetEnvironmentVariables?resource=" + resource;
+
+            var environmentVariableResponse = await LabUserHelper
+                .GetMSIEnvironmentVariablesAsync(uri) 
+                .ConfigureAwait(false);
+
+            //process the response
+            if (!string.IsNullOrEmpty(environmentVariableResponse))
+            {
+                s_envVariables = JsonConvert.DeserializeObject<Dictionary<string, string>>(environmentVariableResponse);
+            }
+            else
+            {
+                return false;
```

Comment:
If this fails, throw an exception. The exception will propagate and fail the test.